.github/workflows/release.yml

```diff
@@ -0,0 +1,58 @@
+name: release
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - lib/version.rb
+
+permissions:
+  contents: write
+  packages: write
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - uses: ruby/setup-ruby@32110d4e311bd8996b2a82bf2a43b714ccc91777 # pin@v1.221.0
+        with:
+          bundler-cache: true
+
+      - name: lint
+        run: bundle exec rubocop -c .rubocop.yml lib/ test/
+
+      - name: test
+        run: bundle exec rake
```

Comment:
Can these be lifted out in some way: by a reusable workflow, or extracted to a script? They are duplicative of two other workflows and I worry that they will get out sync.

Reply:
@bensheldon would you like to adopt this pattern?

https://github.blog/engineering/scripts-to-rule-them-all/

If so, I can add in the scripts that many Ruby projects at GitHub use for bootstrapping, testing, and linting so it is consolidated to a single place.

Reply:
I did exactly this ^ in the following commit: https://github.com/github/rubocop-github/pull/252/commits/6bf8406592739225f59bbbe07cf4448ae28a1191

Let me know what you think

Reply:
Rolled back a few of these changes and just kept it as `script/test` and `script/lint` for now (per our convo)